File: app/src/main/java/me/mitul/aij/helper/HelperSplashScreen.java
package me.mitul.aij.helper;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.readystatesoftware.sqliteasset.SQLiteAssetHelper;

import me.mitul.aij.model.SplashScreen;

public class HelperSplashScreen extends SQLiteAssetHelper {
    public HelperSplashScreen(Context context) {
        super(context, "AIJ_DB.s3db", "/data/data/me.mitul.aij/databases", null, 1);
    }

    public SplashScreen selectSplashTextData(int id) {
        SQLiteDatabase database = getReadableDatabase();
        Cursor cursor = database.rawQuery("SELECT SplashText, TextSize, TextColor FROM MST_Splash WHERE _id = " + id, null);
        SplashScreen splashScreen = new SplashScreen();
        if (cursor.moveToFirst()) {
            splashScreen.setText(cursor.getString(cursor.getColumnIndex("SplashText")));
            splashScreen.setTextSize(cursor.getString(cursor.getColumnIndex("TextSize")));
            splashScreen.setTextColor(cursor.getString(cursor.getColumnIndex("TextColor")));
            return splashScreen;
        }
        cursor.close();
        database.close();
        return null;
    }
}

File: app/src/main/java/me/mitul/aij/helper/HelperSplashScreen.kt
package me.mitul.aij.helper

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import androidx.core.database.getIntOrNull
import androidx.core.database.getStringOrNull
import com.readystatesoftware.sqliteasset.SQLiteAssetHelper
import me.mitul.aij.model.Splash
import me.mitul.aij.utils.Consts

class HelperSplashScreen(context: Context?) : SQLiteAssetHelper(
    context, Consts.DB_NAME, Consts.DB_PATH, null, Consts.DB_VERSION
) {
    private val database: SQLiteDatabase = getReadableDatabase()

    private object Keys {
        const val DB_TBL_NAME = "Splash"
        const val DB_COL_ID = "id"
        const val DB_COL_CONTENT = "content"
    }

    fun getSplashContent(): MutableList<Splash> {
        val cursor = database.rawQuery("SELECT * FROM ${Keys.DB_TBL_NAME}", null)
        val items = mutableListOf<Splash>()
        while (cursor.moveToNext()) {
            items += Splash(
                id = cursor.getIntOrNull(cursor.getColumnIndex(Keys.DB_COL_ID)) ?: 0,
                text = cursor.getStringOrNull(cursor.getColumnIndex(Keys.DB_COL_CONTENT)) ?: ""
            )
        }

        cursor.close()
        database.close()
        return items
    }
}
